SECURITY.md

```diff
@@ -1,19 +1,229 @@
+the [contributing](CONTRIBUTING.md) and [reviewing](REVIEWING.md) documents.
+
+### Policies
+
+Vector maintains this security policy. Changed are communicated to all Vector
+team members.
+
+### Two-factor Authentication
+
+All Vector team members are required to enable two-factor authentication
+on their work machines as well as their Github accounts.
+
+## Development
+
+### Design & Architecture
+
+The base of Vector's security lies in our choice of underlying technology and
+decisions around design and architecture.
 
-## Reporting a Vulnerability
+#### Rust
+
+The [Rust programming language][urls.rust] is memory and thread-safe; it will
+catch many common sources of vulnerabilities at compile time.
+
+#### Unsafe Code
+
+Vector does not allow the use of unsafe code except in circumstances where it
+is required, such as dealing with CFFI.
+
+#### User Privileges
+
+Vector is always designed to run under non-`sudo` privileges, and our
+documentation always defaults to non-`sudo` use.
+
+### Dependencies
+
+Vector aims to reduce the number of dependencies it relies on. If a dependency
+is added it goes through a comprehensive review process that is detailed in
+the [Reviewing guide](REVIEWING.md#dependencies).
+
+### Change Control
+
+As noted above Vector uses the Git version control system on Github.
+
+#### Pull Requests
+
+All changes to Vector must go through a pull request review process.
+
+#### Reviews & Approvals
+
+All pull requests must be reviewed by at least one Vector team member. The
+review process takes into account many factors, all of which are detailed in
+our [Reviewing guide](REVIEWING.md). In exceptional circumstances, this
+approval can be retroactive.
+
+#### Merge Policies
+
+Vector requires pull requests to pass all [automated checks](#automated-checks).
+Once passed, the pull request must be squashed and merged. This creates a clean
+linear history with a Vector team member's co-sign.
+
+#### Automated Checks
+
+When possible, we'll create automated checks to enforce security policies.
+
+##### Vulnerability Scans
+
+Vector implements an automated [`cargo audit` check][urls.cargo_audit]. This
+is part of the [Rust Security advisory database][urls.rust_sec].
+
+##### Fuzz Testing
+
+Vector implements automated fuzz testing to probe our code for other sources
+of potential vulnerabilities.
+
+## Building & Releasing
+
+Vector takes care to secure the build and release process to prevent unintended
+modifications.
+
+### Network Security
+
+All network traffic is secured via TLS and SSH. This includes checking out
+Vector's code from the relevant [protected branch](#protected-branches),
+Docker image retrieval, and publishment of Vector's release artifacts.
+
+### Runtime Isolation
+
+All builds run in an isolated sandbox that is destroyed after each use.
+
+### Asset Encryption
+
+Vector's artifacts are stored on S3 with SSE encryption enabled.
```

Comment:
This seems unnecessary. The artifacts are not confidential, so I don't see how encrypting them on S3 adds anything.